src/main/scala/org/broadinstitute/dsde/workbench/sam/db/tables/ResourceTable.scala

```diff
@@ -18,4 +19,11 @@ object ResourceTable extends SQLSyntaxSupport[ResourceRecord] {
     rs.get(e.name),
     rs.get(e.resourceTypeId)
   )
+
+  def pkQuery(resourceId: ResourceId, resourceTypeName: ResourceTypeName, resourceTableAlias: String = "r"): SQLSyntax = {
+    val r = ResourceTable.syntax(resourceTableAlias)
+    samsqls"""select ${r.id}
+              from ${ResourceTable as r}
+              where ${r.name} = ${resourceId} and ${r.resourceTypeId} = (${ResourceTypeTable.pkQuery(resourceTypeName)})"""
```

Comment:
totally - reading that PostgreSQL perf article you sent last week, we should try to do all this with `JOIN` if we can, rather than sub-queries.  I was thinking about this as I wrote it and sort of wanted to get this working first and if performance was bad, then optimize.  But happy to take a stab at trying to find a modular way to do this with JOINs too

Reply:
I am fine if you want to jira it for now... I satisfied myself that a single PK subquery is fine, I just don't know about another level.